### PR TITLE
API: Rename show_discrete_data argument show to force_show, closes #768

### LINKED ISSUE #768: Change show option to force_show in show_discrete_data
The current `show` doesnt look very nice:

```
x.show(show=True)  # wut?
```

changing it to `force_show` would be more clear.

### PR DESCRIPTION
**Pull request by adler-j**

### CODE REVIEW

File: odl/discr/lp_discr.py [hunk 1]
```diff
@@ -747,9 +747,10 @@ def show(self, title=None, method='', coords=None, indices=None,
             (semantically ``[:, :, shape[2:] // 2]``).
             This option is mutually exclusive to ``coords``.
 
-        show : bool, optional
-            If ``True``, show the plot now. Otherwise, display is deferred
-            deferred until later.
+        force_show : bool, optional
+            If the plot should be forced to show now or deferred until later.
```
Comment (kohr-h) [inline odl/discr/lp_discr.py]: If -> Whether
to show -> to be shown

File: odl/space/base_ntuples.py [hunk 2]
```diff
@@ -432,9 +432,10 @@ def show(self, title=None, method='scatter', show=False, fig=None,
 
             'plot' : graph plot
 
-        show : bool, optional
-            If ``True``, the plot is shown immediately. Otherwise, display is
-            deferred to a later point in time.
+        force_show : bool, optional
+            If the plot should be forced to show now or deferred until later.
+            Note that some backends always displays the plot, regardless of
+            this value.
```
Comment (kohr-h) [inline odl/space/base_ntuples.py]: Same here

File: odl/util/graphics.py [hunk 3]
```diff
@@ -115,9 +116,10 @@ def show_discrete_data(values, grid, title=None, method='',
 
         'wireframe', 'plot_wireframe' : surface plot
 
-
-    show : bool, optional
-        If the plot should be showed now or deferred until later
+    force_show : bool, optional
+        If the plot should be forced to show now or deferred until later.
+        Note that some backends always displays the plot, regardless of this
+        value.
```
Comment (kohr-h) [inline odl/util/graphics.py]: Same here

**Comment by adler-j** [comment]
Fixed comments and rebased from master, merge after CI.

**Pull request merged**